tasks/run_puppet_device.rb

```diff
@@ -42,7 +42,11 @@ def read_device_certificate_fingerprints(cert_name)
   return nil unless certificate
   fingerprints = {}
   fingerprints['default'] = certificate.fingerprint
-  mdas = [:SHA1, :SHA224, :SHA256, :SHA384, :SHA512] # ssl_host.suitable_message_digest_algorithms was removed in Puppet 6, specifying the array directly
+  mdas = if Puppet.version.to_f > 5.99
```

Comment:
This should use a comparison like https://github.com/puppetlabs/puppet-resource_api/blob/1b08a12fa8df13c6a9c0c01b50bd65a8a41acd64/Gemfile#L12 or, even better, check `Puppet.respond_to? :valid_digest_algorithms` (http://ruby-doc.org/core-2.4.0/Object.html#method-i-respond_to-3F) .